### ISSUE TITLE
dart plugin error

### ISSUE THREAD

**Issue by minh98** [dart-lang/sdk#38408]
Analyzer Feedback from IntelliJ

## Version information

- `IDEA AI-191.8026.42.35.5791312`
- `2.5.0-edge.1ef83b86ae637ffe7359173804cbc6d3fa25e6db`
- `AI-191.8026.42.35.5791312, JRE 1.8.0_202-release-1483-b49-5587405x64 JetBrains s.r.o, OS Linux(amd64) v5.0.0-27-generic, screens 1920x1080`

## Exception
```
Dart analysis server, SDK version 2.5.0-edge.1ef83b86ae637ffe7359173804cbc6d3fa25e6db, server version 1.27.2, error: Failed to handle completion domain request: {id: 5855, method: completion.getSuggestions, params: {file: /home/tbm98/Documents/FLUTTER_PROJECTS/xechung_mobile/lib/src/apis/store_api.dart, offset: 50}, clientRequestTime: 1568623646055}
NoSuchMethodError: The getter 'isSynthetic' was called on null.
Receiver: null
Tried calling: isSynthetic
#0      Object.noSuchMethod (dart:core-patch/object_patch.dart:51:5)
#1      isTokenDot (package:analysis_server/src/services/completion/dart/completion_ranking_internal.dart:139:17)
#2      testFollowingDot (package:analysis_server/src/services/completion/dart/completion_ranking_internal.dart:144:31)
#3      CompletionRanking.rerank (package:analysis_server/src/services/completion/dart/completion_ranking.dart:131:33)
<asynchronous suspension>
#4      DartCompletionManager.computeSuggestions (package:analysis_server/src/services/completion/dart/completion_manager.dart:185:35)
<asynchronous suspension>
#5      CompletionDomainHandler.computeSuggestions (package:analysis_server/src/domain_completion.dart:122:42)
<asynchronous suspension>
#6      CompletionDomainHandler.processRequest (package:analysis_server/src/domain_completion.dart:392:5)
<asynchronous suspension>
#7      CompletionDomainHandler.handleRequest.<anonymous closure> (package:analysis_server/src/domain_completion.dart:282:9)
#8      _rootRun (dart:async/zone.dart:1124:13)
#9      _CustomZone.run (dart:async/zone.dart:1021:19)
#10     _runZoned (dart:async/zone.dart:1516:10)
#11     runZoned (dart:async/zone.dart:1500:12)
#12     CompletionDomainHandler.handleRequest (package:analysis_server/src/domain_completion.dart:275:12)
#13     AnalysisServer.handleRequest.<anonymous closure>.<anonymous closure> (package:analysis_server/src/analysis_server.dart:325:45)
#14     _PerformanceTagImpl.makeCurrentWhile (package:analyzer/src/generated/utilities_general.dart:258:15)
#15     AnalysisServer.handleRequest.<anonymous closure> (package:analysis_server/src/analysis_server.dart:321:50)
#16     _rootRun (dart:async/zone.dart:1124:13)
#17     _CustomZone.run (dart:async/zone.dart:1021:19)
#18     _runZoned (dart:async/zone.dart:1516:10)
#19     runZoned (dart:async/zone.dart:1500:12)
#20     AnalysisServer.handleRequest (package:analysis_server/src/analysis_server.dart:320:5)
#21     ByteStreamServerChannel._readRequest.<anonymous closure> (package:analysis_server/src/channel/byte_stream_channel.dart:174:16)
#22     _PerformanceTagImpl.makeCurrentWhile (package:analyzer/src/generated/utilities_general.dart:258:15)
#23     ByteStreamServerChannel._readRequest (package:analysis_server/src/channel/byte_stream_channel.dart:165:47)
#24     ByteStreamServerChannel.listen.<anonymous closure> (package:analysis_server/src/channel/byte_stream_channel.dart:110:26)
#25     _rootRunUnary (dart:async/zone.dart:1132:38)
#26     _CustomZone.runUnary (dart:async/zone.dart:1029:19)
#27     _CustomZone.runUnaryGuarded (dart:async/zone.dart:931:7)
#28     _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:336:11)
#29     _BufferingStreamSubscription._add (dart:async/stream_impl.dart:263:7)
#30     _SinkTransformerStreamSubscription._add (dart:async/stream_transformers.dart:68:11)
#31     _EventSinkWrapper.add (dart:async/stream_transformers.dart:15:11)
#32     _StringAdapterSink.add (dart:convert/string_conversion.dart:236:11)
#33     _LineSplitterSink._addLines (dart:convert/line_splitter.dart:150:13)
#34     _LineSplitterSink.addSlice (dart:convert/line_splitter.dart:125:5)
#35     StringConversionSinkMixin.add (dart:convert/string_conversion.dart:163:5)
#36     _SinkTransformerStreamSubscription._handleData (dart:async/stream_transformers.dart:120:24)
#37     _rootRunUnary (dart:async/zone.dart:1132:38)
#38     _CustomZone.runUnary (dart:async/zone.dart:1029:19)
#39     _CustomZone.runUnaryGuarded (dart:async/zone.dart:931:7)
#40     _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:336:11)
#41     _BufferingStreamSubscription._add (dart:async/stream_impl.dart:263:7)
#42     _SinkTransformerStreamSubscription._add (dart:async/stream_transformers.dart:68:11)
#43     _EventSinkWrapper.add (dart:async/stream_transformers.dart:15:11)
#44     _StringAdapterSink.add (dart:convert/string_conversion.dart:236:11)
#45     _StringAdapterSink.addSlice (dart:convert/string_conversion.dart:241:7)
#46     _Utf8ConversionSink.addSlice (dart:convert/string_conversion.dart:312:20)
#47     _Utf8ConversionSink.add (dart:convert/string_conversion.dart:305:5)
#48     _ConverterStreamEventSink.add (dart:convert/chunked_conversion.dart:72:18)
#49     _SinkTransformerStreamSubscription._handleData (dart:async/stream_transformers.dart:120:24)
#50     _rootRunUnary (dart:async/zone.dart:1132:38)
#51     _CustomZone.runUnary (dart:async/zone.dart:1029:19)
#52     _CustomZone.runUnaryGuarded (dart:async/zone.dart:931:7)
#53     _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:336:11)
#54     _BufferingStreamSubsc...
```

For additional log information, please append the contents of
file:///tmp/report.txt.
contents:

```
Dart analysis server, SDK version 2.5.0-edge.1ef83b86ae637ffe7359173804cbc6d3fa25e6db, server version 1.27.2, error: Failed to handle completion domain request: {id: 5855, method: completion.getSuggestions, params: {file: /home/tbm98/Documents/FLUTTER_PROJECTS/xechung_mobile/lib/src/apis/store_api.dart, offset: 50}, clientRequestTime: 1568623646055}
NoSuchMethodError: The getter 'isSynthetic' was called on null.
Receiver: null
Tried calling: isSynthetic
#0      Object.noSuchMethod (dart:core-patch/object_patch.dart:51:5)
#1      isTokenDot (package:analysis_server/src/services/completion/dart/completion_ranking_internal.dart:139:17)
#2      testFollowingDot (package:analysis_server/src/services/completion/dart/completion_ranking_internal.dart:144:31)
#3      CompletionRanking.rerank (package:analysis_server/src/services/completion/dart/completion_ranking.dart:131:33)
<asynchronous suspension>
#4      DartCompletionManager.computeSuggestions (package:analysis_server/src/services/completion/dart/completion_manager.dart:185:35)
<asynchronous suspension>
#5      CompletionDomainHandler.computeSuggestions (package:analysis_server/src/domain_completion.dart:122:42)
<asynchronous suspension>
#6      CompletionDomainHandler.processRequest (package:analysis_server/src/domain_completion.dart:392:5)
<asynchronous suspension>
#7      CompletionDomainHandler.handleRequest.<anonymous closure> (package:analysis_server/src/domain_completion.dart:282:9)
#8      _rootRun (dart:async/zone.dart:1124:13)
#9      _CustomZone.run (dart:async/zone.dart:1021:19)
#10     _runZoned (dart:async/zone.dart:1516:10)
#11     runZoned (dart:async/zone.dart:1500:12)
#12     CompletionDomainHandler.handleRequest (package:analysis_server/src/domain_completion.dart:275:12)
#13     AnalysisServer.handleRequest.<anonymous closure>.<anonymous closure> (package:analysis_server/src/analysis_server.dart:325:45)
#14     _PerformanceTagImpl.makeCurrentWhile (package:analyzer/src/generated/utilities_general.dart:258:15)
#15     AnalysisServer.handleRequest.<anonymous closure> (package:analysis_server/src/analysis_server.dart:321:50)
#16     _rootRun (dart:async/zone.dart:1124:13)
#17     _CustomZone.run (dart:async/zone.dart:1021:19)
#18     _runZoned (dart:async/zone.dart:1516:10)
#19     runZoned (dart:async/zone.dart:1500:12)
#20     AnalysisServer.handleRequest (package:analysis_server/src/analysis_server.dart:320:5)
#21     ByteStreamServerChannel._readRequest.<anonymous closure> (package:analysis_server/src/channel/byte_stream_channel.dart:174:16)
#22     _PerformanceTagImpl.makeCurrentWhile (package:analyzer/src/generated/utilities_general.dart:258:15)
#23     ByteStreamServerChannel._readRequest (package:analysis_server/src/channel/byte_stream_channel.dart:165:47)
#24     ByteStreamServerChannel.listen.<anonymous closure> (package:analysis_server/src/channel/byte_stream_channel.dart:110:26)
#25     _rootRunUnary (dart:async/zone.dart:1132:38)
#26     _CustomZone.runUnary (dart:async/zone.dart:1029:19)
#27     _CustomZone.runUnaryGuarded (dart:async/zone.dart:931:7)
#28     _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:336:11)
#29     _BufferingStreamSubscription._add (dart:async/stream_impl.dart:263:7)
#30     _SinkTransformerStreamSubscription._add (dart:async/stream_transformers.dart:68:11)
#31     _EventSinkWrapper.add (dart:async/stream_transformers.dart:15:11)
#32     _StringAdapterSink.add (dart:convert/string_conversion.dart:236:11)
#33     _LineSplitterSink._addLines (dart:convert/line_splitter.dart:150:13)
#34     _LineSplitterSink.addSlice (dart:convert/line_splitter.dart:125:5)
#35     StringConversionSinkMixin.add (dart:convert/string_conversion.dart:163:5)
#36     _SinkTransformerStreamSubscription._handleData (dart:async/stream_transformers.dart:120:24)
#37     _rootRunUnary (dart:async/zone.dart:1132:38)
#38     _CustomZone.runUnary (dart:async/zone.dart:1029:19)
#39     _CustomZone.runUnaryGuarded (dart:async/zone.dart:931:7)
#40     _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:336:11)
#41     _BufferingStreamSubscription._add (dart:async/stream_impl.dart:263:7)
#42     _SinkTransformerStreamSubscription._add (dart:async/stream_transformers.dart:68:11)
#43     _EventSinkWrapper.add (dart:async/stream_transformers.dart:15:11)
#44     _StringAdapterSink.add (dart:convert/string_conversion.dart:236:11)
#45     _StringAdapterSink.addSlice (dart:convert/string_conversion.dart:241:7)
#46     _Utf8ConversionSink.addSlice (dart:convert/string_conversion.dart:312:20)
#47     _Utf8ConversionSink.add (dart:convert/string_conversion.dart:305:5)
#48     _ConverterStreamEventSink.add (dart:convert/chunked_conversion.dart:72:18)
#49     _SinkTransformerStreamSubscription._handleData (dart:async/stream_transformers.dart:120:24)
#50     _rootRunUnary (dart:async/zone.dart:1132:38)
#51     _CustomZone.runUnary (dart:async/zone.dart:1029:19)
#52     _CustomZone.runUnaryGuarded (dart:async/zone.dart:931:7)
#53     _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:336:11)
#54     _BufferingStreamSubscription._add (dart:async/stream_impl.dart:263:7)
#55     _SyncStreamControllerDispatch._sendData (dart:async/stream_controller.dart:764:19)
#56     _StreamController._add (dart:async/stream_controller.dart:640:7)
#57     _StreamController.add (dart:async/stream_controller.dart:586:5)
#58     _Socket._onData (dart:io-patch/socket_patch.dart:1791:41)
#59     _rootRunUnary (dart:async/zone.dart:1136:13)
#60     _CustomZone.runUnary (dart:async/zone.dart:1029:19)
#61     _CustomZone.runUnaryGuarded (dart:async/zone.dart:931:7)
#62     _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:336:11)
#63     _BufferingStreamSubscription._add (dart:async/stream_impl.dart:263:7)
#64     _SyncStreamControllerDispatch._sendData (dart:async/stream_controller.dart:764:19)
#65     _StreamController._add (dart:async/stream_controller.dart:640:7)
#66     _StreamController.add (dart:async/stream_controller.dart:586:5)
#67     new _RawSocket.<anonymous closure> (dart:io-patch/socket_patch.dart:1339:33)
#68     _NativeSocket.issueReadEvent.issue (dart:io-patch/socket_patch.dart:860:14)
#69     _microtaskLoop (dart:async/schedule_microtask.dart:41:21)
#70     _startMicrotaskLoop (dart:async/schedule_microtask.dart:50:5)
#71     _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:116:13)
#72     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:173:5)

```
1568623646058 <= {"event":"analysis.errors","params":{"file":"/home/tbm98/Documents/FLUTTER_PROJECTS/xechung_mobile/lib/src/apis/store_api.dart","errors":[]}}
1568623646059 <= {"event":"completion.availableSuggestions","params":{"changedLibraries":[{"id":971,"uri":"package:xechung_mobile/src/apis/store_api.dart","items":[{"label":"StoreApi","declaringLibrar
1568623646060 <= {"event":"server.status","params":{"analysis":{"isAnalyzing":false}}}
1568623646060 <= {"event":"analysis.implemented","params":{"file":"/home/tbm98/Documents/FLUTTER_PROJECTS/xechung_mobile/lib/src/apis/store_api.dart","classes":[],"members":[]}}
1568623646060 <= {"event":"analysis.highlights","params":{"file":"/home/tbm98/Documents/FLUTTER_PROJECTS/xechung_mobile/lib/src/apis/store_api.dart","regions":[{"type":"KEYWORD","offset":9,"length":5}
1568623646060 <= {"event":"analysis.navigation","params":{"file":"/home/tbm98/Documents/FLUTTER_PROJECTS/xechung_mobile/lib/src/apis/store_api.dart","regions":[{"offset":15,"length":8,"targets":[0]},{
1568623646060 <= {"event":"analysis.closingLabels","params":{"file":"/home/tbm98/Documents/FLUTTER_PROJECTS/xechung_mobile/lib/src/apis/store_api.dart","labels":[]}}
1568623646060 <= {"event":"analysis.outline","params":{"file":"/home/tbm98/Documents/FLUTTER_PROJECTS/xechung_mobile/lib/src/apis/store_api.dart","kind":"LIBRARY","outline":{"element":{"kind":"COMPILA
1568623646060 <= {"event":"analysis.overrides","params":{"file":"/home/tbm98/Documents/FLUTTER_PROJECTS/xechung_mobile/lib/src/apis/store_api.dart","overrides":[]}}
1568623646061 <= {"event":"flutter.outline","params":{"file":"/home/tbm98/Documents/FLUTTER_PROJECTS/xechung_mobile/lib/src/apis/store_api.dart","outline":{"kind":"DART_ELEMENT","offset":0,"length":12
1568623646450 => {"id":"5856","method":"analysis.updateContent","params":{"files":{"/home/tbm98/Documents/FLUTTER_PROJECTS/xechung_mobile/lib/src/apis/store_api.dart":{"type":"add","content":"abstract
1568623646452 <= {"id":"5856","result":{}}
1568623646452 <= {"event":"server.status","params":{"analysis":{"isAnalyzing":true}}}
1568623646454 => {"id":"5857","method":"edit.getAssists","params":{"file":"/home/tbm98/Documents/FLUTTER_PROJECTS/xechung_mobile/lib/src/apis/store_api.dart","offset":51,"length":0},"clientRequestTime
1568623646456 <= {"event":"analysis.errors","params":{"file":"/home/tbm98/Documents/FLUTTER_PROJECTS/xechung_mobile/lib/src/apis/store_api.dart","errors":[]}}
1568623646456 <= {"event":"completion.availableSuggestions","params":{"changedLibraries":[{"id":971,"uri":"package:xechung_mobile/src/apis/store_api.dart","items":[{"label":"StoreApi","declaringLibrar
1568623646458 <= {"id":"5857","result":{"assists":[{"message":"Convert to block documentation comment","edits":[{"file":"/home/tbm98/Documents/FLUTTER_PROJECTS/xechung_mobile/lib/src/apis/store_api.da
1568623646458 <= {"event":"analysis.highlights","params":{"file":"/home/tbm98/Documents/FLUTTER_PROJECTS/xechung_mobile/lib/src/apis/store_api.dart","regions":[{"type":"KEYWORD","offset":9,"length":5}
1568623646458 <= {"event":"analysis.navigation","params":{"file":"/home/tbm98/Documents/FLUTTER_PROJECTS/xechung_mobile/lib/src/apis/store_api.dart","regions":[{"offset":15,"length":8,"targets":[0]},{
1568623646458 <= {"event":"analysis.closingLabels","params":{"file":"/home/tbm98/Documents/FLUTTER_PROJECTS/xechung_mobile/lib/src/apis/store_api.dart","labels":[]}}
1568623646459 <= {"event":"analysis.outline","params":{"file":"/home/tbm98/Documents/FLUTTER_PROJECTS/xechung_mobile/lib/src/apis/store_api.dart","kind":"LIBRARY","outline":{"element":{"kind":"COMPILA
1568623646459 <= {"event":"analysis.overrides","params":{"file":"/home/tbm98/Documents/FLUTTER_PROJECTS/xechung_mobile/lib/src/apis/store_api.dart","overrides":[]}}
1568623646459 <= {"event":"flutter.outline","params":{"file":"/home/tbm98/Documents/FLUTTER_PROJECTS/xechung_mobile/lib/src/apis/store_api.dart","outline":{"kind":"DART_ELEMENT","offset":0,"length":12
1568623646461 <= {"event":"server.status","params":{"analysis":{"isAnalyzing":false}}}
1568623646461 <= {"event":"analysis.implemented","params":{"file":"/home/tbm98/Documents/FLUTTER_PROJECTS/xechung_mobile/lib/src/apis/store_api.dart","classes":[],"members":[]}}
1568623646461 => {"id":"5858","method":"edit.getAssists","params":{"file":"/home/tbm98/Documents/FLUTTER_PROJECTS/xechung_mobile/lib/src/apis/store_api.dart","offset":51,"length":0},"clientRequestTime
1568623646462 => {"id":"5859","method":"edit.getAssists","params":{"file":"/home/tbm98/Documents/FLUTTER_PROJECTS/xechung_mobile/lib/src/apis/store_api.dart","offset":51,"length":0},"clientRequestTime
1568623646463 <= {"id":"5858","result":{"assists":[{"message":"Convert to block documentation comment","edits":[{"file":"/home/tbm98/Documents/FLUTTER_PROJECTS/xechung_mobile/lib/src/apis/store_api.da
1568623646464 <= {"id":"5859","result":{"assists":[{"message":"Convert to block documentation comment","edits":[{"file":"/home/tbm98/Documents/FLUTTER_PROJECTS/xechung_mobile/lib/src/apis/store_api.da
1568623646480 <= {"event":"server.error","params":{"isFatal":false,"message":"Failed to handle completion domain request: {id: 5855, method: completion.getSuggestions, params: {file: /home/tbm98/Docum
```

**Comment by hieuseesaa** [comment]
`The getter 'isSynthetic' was called on null.` Nó show lỗi null này rồi. fix null thôi.

**Comment by keertip** [comment]
This has been fixed by [118831e](https://github.com/dart-lang/sdk/commit/118831ec535f0c214ac025914d53961a0732d2a2) and fix is available in the latest dev version.

@hieuseesaa , can you upgrade to 2.6.0-dev.0.0 and see if that fixes the issue.

**Comment by minh98** [comment]
yep
